### PR TITLE
new structure and story

### PR DESCRIPTION
**Pull request by EvJack**

### CODE REVIEW

File: contribution/dungeon/layout/structure-mask.json [hunk 1]
```diff
@@ -1245,6 +1245,66 @@
...
+          "#","#"," "," "," "," "," ","#"," "," "," ","#"," "," ","#",
+          "#","#"," "," "," "," "," ","#"," "," "," ","#"," "," ","#",
+          "#","#"," "," "," "," "," ","#"," "," "," ","#"," "," ","#",
+          "#","#"," "," "," "," "," ","#"," "," "," ","#"," "," ","#",
+          "#","#","#","#"," "," "," ","#"," "," "," ","#"," "," "," ",
+          "#","#"," "," ","#"," "," ","#"," "," "," ","#"," "," ","#",
+          "#","#"," "," ","#"," "," ","#"," "," "," ","#"," "," ","#",
+          "#","#"," "," ","#"," "," ","#"," "," "," ","#"," "," ","#",
+          "#","#"," "," ","#"," "," ","#"," "," "," ","#"," "," ","#",
+          "#","#","#","#","#"," ","#","#","#"," "," ","#"," "," ","#",
```
Comment (MagicZDS) [inline contribution/dungeon/layout/structure-mask.json:1263]: Loop on "G" should be avoided, I think

File: contribution/dungeon/layout/structure-mask.json [hunk 2]
```diff
@@ -1245,6 +1245,66 @@
...
+    {
+      "tiles": [
+          "#","#","#","#","#","#","#"," ","#","#","#","#","#","#","#",
+          "#"," "," "," "," "," "," "," "," "," "," "," "," "," ","#",
+          "#"," ","#"," ","#"," ","#"," ","#"," ","#","#","#"," ","#",
+          "#"," ","#"," ","#"," ","#"," ","#"," ","#"," "," ","#","#",
+          "#"," ","#"," ","#"," ","#"," ","#"," ","#"," "," ","#","#",
+          "#"," ","#"," ","#"," ","#"," ","#"," ","#"," "," ","#","#",
+          "#"," ","#"," ","#"," ","#"," ","#"," ","#"," "," ","#","#",
+          " "," ","#","#","#"," ","#"," ","#"," ","#","#","#"," ","#",
```
Comment (MagicZDS) [inline contribution/dungeon/layout/structure-mask.json:1278]: Top-loop on "B" should be avoided, I think

File: contribution/dungeon/layout/structure-mask.json [hunk 3]
```diff
@@ -1245,6 +1245,66 @@
...
+          "#"," ","#"," ","#"," ","#"," ","#"," ","#"," "," ","#","#",
+          "#"," ","#"," ","#"," ","#"," ","#"," ","#"," "," ","#","#",
+          "#"," ","#"," ","#"," ","#"," ","#"," ","#"," "," ","#","#",
+          "#"," ","#"," ","#"," ","#"," ","#"," ","#"," "," ","#","#",
+          "#"," ","#"," ","#"," "," ","#"," "," ","#","#","#"," ","#",
```
Comment (MagicZDS) [inline contribution/dungeon/layout/structure-mask.json:1283]: Bottom-loop on "B" should be avoided, I think

**Review by MagicZDS — CHANGES_REQUESTED** [review]
You should make a hole in loops (1 in "G", 2 in "B"), like it already exists on "D" (see line 1038 on this file)

**Comment by DexterHuang** [comment]
yeah seems like two of the design as closed loop, there cannot be close loop, it will result in blocked area

**Comment by EvJack** [comment]
Fixed bugs

**Pull request merged**